### PR TITLE
Feature/instructions

### PR DESCRIPTION
**Pull request by agh-contributor**

### CODE REVIEW

File: src/main/java/com/ccuk/demo/api/ApplicationController.java [hunk 1]
```diff
@@ -53,6 +54,48 @@ public String testFeatureEndpoint(Principal principal) {
         return enabled ? maintenanceService.welcomeMessage() : FEATURE_NOT_ENABLED_FOR_USER;
     }
 
+    @PostMapping(value = INSTRUCTION_RESOURCE, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (NabilCC) [inline src/main/java/com/ccuk/demo/api/ApplicationController.java:57]: Engineering at its finest

Comment (agh-contributor) [reply src/main/java/com/ccuk/demo/api/ApplicationController.java:57]: Thank you

File: src/main/java/com/ccuk/demo/api/ApplicationController.java [hunk 2]
```diff
@@ -53,6 +54,48 @@ public String testFeatureEndpoint(Principal principal) {
...
+                    .toUri();
+            return ResponseEntity.created(location).build();
+        } catch (ValidationException e) {
+            return ResponseEntity.badRequest().body(invalidRequestMessage(e));
+        }
+    }
+
+    @GetMapping(value = INSTRUCTION_RESOURCE + "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getInstructionById(@PathVariable("id") long id, Principal principal) {
+        if (!featureFlagService.isFeatureEnabledForUser(principal, FeatureFlag.GET_INSTRUCTION_BY_ID)) {
```
Comment (NabilCC) [inline src/main/java/com/ccuk/demo/api/ApplicationController.java:82]: I forgot about this

Comment (NabilCC) [reply src/main/java/com/ccuk/demo/api/ApplicationController.java:82]: OK

Comment (agh-contributor) [reply src/main/java/com/ccuk/demo/api/ApplicationController.java:82]: yes

**Pull request merged**